File: day1/main.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func readInput(filename string) []int {
	filePath := "./" + filename
	data, err := os.ReadFile(filePath)
	check(err)

	raw_lines := strings.Split(string(data), "\n")

	// Convert each string to an int
	var lines []int
	for _, c := range raw_lines {
		value, err := strconv.Atoi(c)
		if err == nil {
			lines = append(lines, value)
		}
	}
	return lines
}

func partA() {
	input := readInput("input.txt")

	// starting with 2nd number, add to count if number is greater than the previous
	count := 0
	for i := 0; i < len(input); i++ {
		// fmt.Print(input[i])
		if i == 0 {
			continue
		}
		if input[i] > input[i-1] {
			count++
			// fmt.Println(" increased - " + strconv.Itoa(count))
		} else {
			// fmt.Println(" decreased - " + strconv.Itoa(count))
		}
	}

	fmt.Println(count)
}

func main() {
	partA()
}

File: day1/main.py
#!/usr/bin/env python3


def read_input(filename):
    with open(f"./{filename}") as f:
        lines = f.read().splitlines()
        return lines


def a(nums):
    count = 0
    for i in range(len(nums)):
        if i == 0:
            continue
        elif nums[i] > nums[i - 1]:
            count += 1
            # print(f"{nums[i]} increased - {count}")
        # else:
        #   print(f"{nums[i]} decreased - {count}")

    print(count)


def b(nums):
    count = 0
    last = 0
    for i in range(len(nums)):
        if i == 0 or i == 1:
            continue

        total = nums[i] + nums[i - 1] + nums[i - 2]
        # print(total)

        if last != 0 and total > last:
            # print("bump")
            count += 1

        last = total
    print(count)


if __name__ == "__main__":
    lines = read_input("input.txt")
    nums = [int(line) for line in lines]
    a(nums)
    b(nums)
